### PR TITLE
[OTE-TEST] Disable obsolete test cases

### PR DESCRIPTION
**Pull request by harimkang**
### This PR includes :
- Remove (Change to skip) some model template test cases for OTE-CI test (some external backend's API & performance tests)
- Fixed some library version conflict issue (in OTE CI venv)
- Fixed some test cases not working (MPA ote_cli tests)
- Add MPA pre-commit tests (not for pydocstyle -> will check it out later)
- Resolve prettier issue (data/car_tree_bug/annotations/hierarchical_default.json)
- Resolve black issue (ote_cli/ote_cli/utils/hpo.py)

### Result of this PR
- validation-report: http://validationreports.sclab.intel.com:8006/reports/build_number_report?test_session_build_number=pr-ote-test-590&environment=iotg-dev-workstation-17

### CODE REVIEW

**Comment by harimkang** [comment]
run ote-test

**Comment by harimkang** [comment]
run ote-test

**Comment by harimkang** [comment]
run ote-test

**Comment by harimkang** [comment]
run ote-test

**Comment by harimkang** [comment]
run ote-test

**Comment by harimkang** [comment]
run ote-test

**Comment by harimkang** [comment]
run ote-test

**Comment by harimkang** [comment]
run ote-test

**Comment by harimkang** [comment]
run ote-test

**Comment by harimkang** [comment]
run ote-test

**Comment by harimkang** [comment]
run ote-test

**Comment by harimkang** [comment]
run ote-test

**Comment by harimkang** [comment]
run ote-test

**Comment by harimkang** [comment]
run ote-test

**Comment by harimkang** [comment]
run ote-test

**Comment by harimkang** [comment]
run ote-test

**Comment by harimkang** [comment]
run ote-test

**Comment by harimkang** [comment]
run ote-test

**Comment by harimkang** [comment]
run ote-test

**Comment by harimkang** [comment]
run ote-test

**Comment by harimkang** [comment]
run ote-test

**Comment by harimkang** [comment]
run ote-test

**Comment by harimkang** [comment]
run ote-test

**Comment by harimkang** [comment]
run ote-test

**Comment by harimkang** [comment]
run ote-test

**Comment by harimkang** [comment]
run ote-test

**Comment by harimkang** [comment]
run ote-test

**Comment by harimkang** [comment]
run ote-test

**Comment by harimkang** [comment]
run ote-test

**Comment by harimkang** [comment]
run ote-test

**Comment by harimkang** [comment]
run ote-test

File: external/model-preparation-algorithm/init_venv.sh [hunk 1]
```diff
@@ -128,12 +128,6 @@ pip install -e ../mmdetection/submodule || exit 1
 pip install -e ../mmsegmentation/submodule || exit 1
 pip install -e submodule || exit 1
 
-MPA_DIR=`realpath submodule`
-echo "export MPA_DIR=${MPA_DIR}" >> ${venv_dir}/bin/activate
-
-# Install OTE CLI
-pip install -e ../../ote_cli || exit 1
```
Comment (JihwanEom) [inline external/model-preparation-algorithm/init_venv.sh:135]: Could you explain why we should remove these lines?

File: external/model-preparation-algorithm/tests/ote_cli/test_detection.py [hunk 2]
```diff
@@ -46,7 +44,7 @@
         '--learning_parameters.num_iters',
         '2',
         '--learning_parameters.batch_size',
-        '2'
+        '4'
```
Comment (JihwanEom) [inline external/model-preparation-algorithm/tests/ote_cli/test_detection.py:47]: Why did you change batch size?

File: external/model-preparation-algorithm/tests/ote_cli/test_segmentation.py [hunk 3]
```diff
@@ -50,18 +50,18 @@
...
+        '4'
     ]
 }
 
-root = '/tmp/ote_cli_seg/'
+root = '/tmp/ote_cli/'
 ote_dir = os.getcwd()
 
-templates = Registry('external/model-preparation-algorithm', experimental=True).filter(task_type='SEGMENTATION').templates
+templates = Registry('external/model-preparation-algorithm').filter(task_type='SEGMENTATION').templates
```
Comment (JihwanEom) [inline external/model-preparation-algorithm/tests/ote_cli/test_segmentation.py:60]: Hmm, lite-hrnet-18 will be collected. (not mod-2). How do you think about this?

File: external/model-preparation-algorithm/tests/ote_cli/test_detection.py (hunk 2 above)
Comment (harimkang) [reply external/model-preparation-algorithm/tests/ote_cli/test_detection.py:47]: I modified it the same as other tasks.

File: external/model-preparation-algorithm/init_venv.sh (hunk 1 above)
Comment (harimkang) [reply external/model-preparation-algorithm/init_venv.sh:135]: This causes library conflict during MPA environment configuration of API & performance tests in OTE CI test. In addition, init_venv does not install the ote_cli for other backends. so align this

I removed it for the above two reasons.
In guide doc, this is a guide to install ote_cli after init_venv, so I think it will be okay.

File: external/model-preparation-algorithm/tests/ote_cli/test_segmentation.py (hunk 3 above)
Comment (harimkang) [reply external/model-preparation-algorithm/tests/ote_cli/test_segmentation.py:60]: I'm sorry I didn't understand your comment.

Comment (JihwanEom) [reply external/model-preparation-algorithm/tests/ote_cli/test_segmentation.py:60]: Please refer here : https://github.com/openvinotoolkit/training_extensions/tree/develop/external/model-preparation-algorithm/configs/segmentation/ocr-lite-hrnet-18
Lite-hrnet-18/template.yaml is reverted. From this, lite-hrnet-18 will be collected as template even though we are keeping only for compatibility purpose.

Comment (harimkang) [reply external/model-preparation-algorithm/tests/ote_cli/test_segmentation.py:60]: I think compatibility can be established by passing this test necessarily. What do you think?

Comment (harimkang) [reply external/model-preparation-algorithm/tests/ote_cli/test_segmentation.py:60]: Also, I know that the template, which is experimental, is not currently in the segmentation config.

**Review by goodsong81 — APPROVED** [review]
LGTM. What a job!

**Review by JihwanEom — APPROVED** [review]
LGTM! Thanks for your effort!

**Comment by harimkang** [comment]
validation-report: http://validationreports.sclab.intel.com:8006/reports/build_number_report?test_session_build_number=pr-ote-test-599&environment=iotg-dev-workstation-17
This PR is for OTE-CI to work properly, so i will merge it here and solve the FAILED test in other PRs.
